Python/parking_lot.c

```diff
@@ -102,7 +102,13 @@ _PySemaphore_PlatformWait(_PySemaphore *sema, PyTime_t timeout)
         millis = INFINITE;
     }
     else {
```

Comment:
I would suggest to use _PyTime_AsMilliseconds(_PyTime_ROUND_TIMEOUT), and then clamp to DWORD_MAX.

_PyTime_AsMilliseconds() rounds away from zero which can give a different result than your naive division. For example, 1 nanosecond is rounded to 1 ms instead of 0 ms.

Reply:
Done!